hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -282,20 +283,22 @@ private static void processRollbackMetadata(HoodieActiveTimeline metadataTableTi
     List<HoodieRecord> records = new LinkedList<>();
     int[] fileChangeCount = {0, 0}; // deletes, appends
 
-    partitionToDeletedFiles.forEach((partition, deletedFiles) -> {
+    partitionToDeletedFiles.forEach((partitionName, deletedFiles) -> {
       fileChangeCount[0] += deletedFiles.size();
+      final String partition = partitionName.equals("") ? NON_PARTITIONED_NAME : partitionName;
 
       Option<Map<String, Long>> filesAdded = Option.empty();
-      if (partitionToAppendedFiles.containsKey(partition)) {
-        filesAdded = Option.of(partitionToAppendedFiles.remove(partition));
+      if (partitionToAppendedFiles.containsKey(partitionName)) {
```

Comment:
This is prone to error when refactoring the code. This depends on the code ordering of the partitionToDeletedFiles and partitionToAppendedFiles. Instead we can remap the keys for both at once and avoid all these back and forth lookups.

Reply:
again, we create HoodieRecords for metadata table within forEach, and so couldn't do it outside.